eng/WpfArcadeSdk/tools/InjectModuleInitializer.targets

```diff
@@ -0,0 +1,423 @@
+
+      <!--/ITANIUM                Target processor: Intel Itanium-->
+      <Itanium ParameterType="System.String" Required="false" />
+
+      <!--/X64                    Target processor: 64bit AMD processor-->
+      <X64 ParameterType="System.String" Required="false" />
+
+      <!--/ARM                    Target processor: ARM processor-->
+      <Arm ParameterType="System.String" Required="false" />
+
+      <!--/32BITPREFERRED         Create a 32BitPreferred image (PE32)-->
+      <Prefer32Bit ParameterType="System.String" Required="false" />
+
+      <!--/ENC=<file>             Create Edit-and-Continue deltas from specified source file-->
+      <ENC ParameterType="System.String" Required="false" />
+
+    </ParameterGroup>
+    <Task>
+      <Code Type="Class" Language="cs" Source="$(WpfArcadeSdkToolsDir)InjectModuleInitializer\ILAsmTask.cs" />
+    </Task>
+  </UsingTask>
+
+  <!-- Add a module initializer to the supplied IL file. -->
+  <UsingTask TaskName="AddModuleConstructorTask"
+             TaskFactory="$(TaskFactory)"
+             AssemblyFile="$(TaskFactoryAssemblyFile)">
+    <ParameterGroup>
+      <ILFile ParameterType="System.String" Required="true" Output="true"/>
+      <MsCorLibAssemblySectionIL ParameterType="System.String" Required="true" Output="true"/>
+      <ModuleConstructorIL ParameterType="System.String" Required="true" Output="true"/>
+    </ParameterGroup>
+    <Task>
+      <Code Type="Class" Language="cs" Source="$(WpfArcadeSdkToolsDir)InjectModuleInitializer\AddModuleConstructorTask.cs" />
+    </Task>
+  </UsingTask>
+
+  <!-- 
+       In order to run ILDAsm and ILAsm from .NET Core, we need both the IL* executables and the corresponding version 
+       of CoreCLR in the same directory.  This target ensures that for every project using module initializer
+       injection.
+  -->
+  <Target Name="CopyILTools" BeforeTargets="DeconstructDll" Condition="'$(UseNetCoreILTools)'=='true'">
+    <ItemGroup>
+      <ILDAsm_Files Include="$(ILDAsmPath)*"/>
+      <ILAsm_Files Include="$(ILAsmPath)*"/>
+      <CoreCLR_Files Include="$(CoreCLRPath)*"/>
+    </ItemGroup>
+
+    <MakeDir Directories="$(ILToolsPath)" Condition="!Exists($(ILToolsPath))"/>
+
+    <Copy SourceFiles="@(ILDAsm_Files)" DestinationFolder="$(ILToolsPath)" SkipUnchangedFiles="true"/>
+    <Copy SourceFiles="@(ILAsm_Files)" DestinationFolder="$(ILToolsPath)" SkipUnchangedFiles="true"/>
+    <Copy SourceFiles="@(CoreCLR_Files)" DestinationFolder="$(ILToolsPath)" SkipUnchangedFiles="true"/>
+  </Target>
+
+  <!--
+      Locate the IL tools included with the highest version of .NET Framework and the .NET Framework SDK installed on the machine.
+  -->
+  <Target Name="FindNetFrameworkILTools" BeforeTargets="DeconstructDll" Condition="'$(UseNetCoreILTools)'!='true'">
+    <PropertyGroup>
+      <DotNetFrameworkBitness Condition="'$(Platform)'!='AnyCPU' and '$(Platform)'!='Win32'">Bitness64</DotNetFrameworkBitness>
```

Comment:
platform detection - see previous comments. 

In fact, we already define `$(Architecture)` in props...